src/helperFunctions/compare_sets.py

```diff
@@ -1,7 +1,16 @@
-from typing import Iterable, List, T, Tuple
+from typing import Iterable, Iterator, List, Tuple, TypeVar
 
+# Generics (not intended for export)
+_T = TypeVar('_T')
```

Comment:
Why is this useful or rather more useful then just using `list` in the definitions above?

Reply:
Without this, it would not be clear that the output list contains objects of the same type as the input (also for the IDE)